enterprise-suite/templates/es-console.yaml

```diff
@@ -15,6 +15,13 @@ spec:
         runAsUser: {{ .Values.podUID }}
       {{ end }}
       containers:
+      - name: dnsmasq
+        image: "janeczku/go-dnsmasq:release-1.0.7"
```

Comment:
Is there a reason to not use dnsmasq? This golang version looks relatively new and not widely used. It's also unmaintained - the last commit was 2 years ago. I would trust the native dnsmasq a lot more.

Reply:
Also thinking in terms of: security issues, redhat certification. Using official dnsmasq will better meet these needs.

Reply:
Good points. This one was instead of me writing an init container to parse /etc/resolv.conf, but maybe we should think about that option. Needing it is totally silly because nginx requires an external resolver in the mode where it does lookups at run time, but what we want is just use the built in resolver, no special resolver needed.

I think the go-dnsmasq repo could also be inactive because it is "done" meaning there's nothing else needed, it is truly simple. But yeah, not a great look since at least there could have been a go runtime update along the way.

In other words, all this does is provide a go runtime that does a dns lookup, as a dns service.

I was tempted to use "kube-dns.kube-system" as the "resolver" as an example, but steered off of it as there seems a bunch of work in this area with non-kube-dns alternatives emerging.

One alternative is to have an init container that parses the resolver out of /etc/resolv.conf and gives that to nginx in a config file.

Reply:
@jsravn what do you say to the idea of going forward with this now, make a follow on ticket to noodle out how to improve the external resolver situation?  The point being that this fixes a bunch of problems we're having pretty regularly, and the resolver thing is quite fiddly / needs some care.

Reply:
Ah okay, so go-dnsmasq actually reads the search domains out of `resolv.conf`. I missed that. Yeah, I don't think there would be a good way to handle this with dnsmasq.

Like you said, you could just use the kube-dns resolver - but that could bypass a customer's custom setup, like if they wanted to have local dnsmasq caching things on the host, or one of the other options. `/etc/resolv.conf` is the handoff point for apps - so we would have to use that one way or another to find the correct DNS server to use and search domains. It's pretty irritating that nginx doesn't support /etc/resolv.conf out of the box.